src/SFA.DAS.Assessor.Functions/Domain/Learners/EnqueueLearnerInfoCommand.cs

```diff
@@ -0,0 +1,107 @@
+    {
+        public ICollector<string> StorageQueue { get; set; }
+
+        private readonly IOuterApiClient _outerApiClient;
+        private readonly ILogger<EnqueueLearnerInfoCommand> _logger;
+        private readonly IAssessorServiceRepository _assessorServiceRepository;
+
+        public EnqueueLearnerInfoCommand(IOuterApiClient outerApiClient,
+            ILogger<EnqueueLearnerInfoCommand> logger,
+            IAssessorServiceRepository assessorServiceRepository)
+        {
+            _outerApiClient = outerApiClient;
+            _logger = logger;
+            _assessorServiceRepository = assessorServiceRepository;
+        }
+
+        public async Task Execute()
+        {
+            try
+            {
+                _logger.LogInformation("EnqueueLearnerInfoCommand started");
+
+                var learnersToProcessUln = await _assessorServiceRepository.GetLearnersWithoutEmployerInfo();
+
+                if (learnersToProcessUln == null || !learnersToProcessUln.Any())
+                {
+                    _logger.LogInformation($"There is no learner to process EnqueueLearnerInfoCommand completed");
+                    return;
+                }
+
+                _logger.LogInformation($"Learners to process {learnersToProcessUln.Count}");
+
+                //1. Get all Learners From Approvals in batches
+                DateTime? extractStartTime = new DateTime(2000, 1, 1);
+                const int batchSize = 1000;
+                int batchNumber = 0;
+                int count = 0;
+                GetAllLearnersResponse learnersBatch = null;
+
+                do
+                {
+                    batchNumber++;
+                    try
+                    {
+                        learnersBatch = await _outerApiClient.Get<GetAllLearnersResponse>(new GetAllLearnersRequest(extractStartTime, batchNumber, batchSize));
+                        if (learnersBatch?.Learners == null)
+                        {
+                            _logger.LogWarning($"Failed to get learners batch: sinceTime={extractStartTime?.ToString("o", System.Globalization.CultureInfo.InvariantCulture)} batchNumber={batchNumber} batchSize={batchSize}");
+                        }
+
+                        _logger.LogInformation($"Approvals batch import loop. Starting batch {batchNumber} of {learnersBatch.TotalNumberOfBatches}");
+
+                        // 2. Check if the learner needs to be processed if yes enqueue
+                        foreach (var learner in learnersBatch.Learners)
+                        {
+                            long uln = 0;
+                            if (!long.TryParse(learner.ULN, out uln))
+                                continue;
+
+                            int trainingCode = 0;
+                            int.TryParse(learner.TrainingCode, out trainingCode);
+
+                            if (learnersToProcessUln.Contains(uln))
```

Comment:
This only runs once so we may not be concerned about performance - however when it does run it could have several hundred thousand entries in this list and it may take a long time to search a list - should consider building a dictionary from the list

Reply:
Actually maybe we are concerned about performance as each function has a timeout - so it may get terminated before this completed if it takes ages to queue all these messages.

Reply:
As discussed, moving to simple HttpTrigger to drop message and then QueueTrigger to perform Enqueue activity

Reply:
changed